tools/module/generate.py

```diff
@@ -137,16 +139,7 @@ def generate_cpp(parts):
 
             on<Configuration>("{className}.yaml").then([this](const Configuration& cfg) {{
                 // Use configuration here from file {className}.yaml
-
-                // clang-format off
-                auto lvl = cfg["log_level"].as<std::string>();
-                if (lvl == "TRACE") {{ this->log_level = NUClear::TRACE; }}
-                else if (lvl == "DEBUG") {{ this->log_level = NUClear::DEBUG; }}
-                else if (lvl == "INFO") {{ this->log_level = NUClear::INFO; }}
-                else if (lvl == "WARN") {{ this->log_level = NUClear::WARN; }}
-                else if (lvl == "ERROR") {{ this->log_level = NUClear::ERROR; }}
-                else if (lvl == "FATAL") {{ this->log_level = NUClear::FATAL; }}
-                // clang-format on
+                this->log_level = cfg["log_level"].as<utility::support::LogLevel>();
```

Comment:
If you setup the converter to do NUClear::LogLevel instances directly you don't need the .as<>